src/commands/generate/__tests__/generatorTest.ts

```diff
@@ -109,31 +106,44 @@ export { integrationSecond };
 
 const irDir = path.join(__dirname, "../../../../build/ir-test-cases");
 const testCaseDir = path.join(__dirname, "resources/test-cases");
+const flavoredTestCaseDir = path.join(__dirname, "resources/flavored-test-cases");
+const readonlyTestCaseDir = path.join(__dirname, "resources/readonly-test-cases");
 
 describe("definitionTests", () => {
-    let tempDir: string;
-
-    beforeEach(() => {
-        tempDir = directory();
-    });
-
     for (const fileName of fs.readdirSync(irDir)) {
```

Comment:
These tests run the generator and compare against the previously generated files in `__tests__/resources/test-cases`. It still does that, but I've now also separated out `flavored-test-cases` so we're still testing the default correctly, and (new) `readonly-test-cases`.